toolchain: update to latest nightly

rmsyn opened the pull request:
Nightly builds compile with the current version release (2023-05-15).

Adding `codegen-units = 1` fixes compilation errors for debug builds.

See: https://github.com/rust-lang/compiler-builtins/issues/523#issuecomment-1547339559

rmsyn committed:
toolchain: update to latest nightly

Nightly builds compile with the current version release (2023-05-15).

Adding `codegen-units = 1` fixes compilation errors for debug builds.

See: https://github.com/rust-lang/compiler-builtins/issues/523#issuecomment-1547339559

agausmann commented on Cargo.toml:
```diff
@@ -18,6 +18,7 @@ features = ["arduino-leonardo"]
 [profile.dev]
 opt-level = "s"
 lto = true
+codegen-units = 1
```
Let's put a comment here that references the [upstream issue](https://github.com/rust-lang/compiler-builtins/issues/523#issuecomment-1547339559)

agausmann commented:
I can't compile with this version:

```
$ cargo +nightly-2023-05-15 b --release --example arduino_keyboard
...
LLVM ERROR: Expected a constant shift amount!
```

`nightly-2023-03-24` works fine however.

rmsyn commented:
> I can't compile with this version

:facepalm: I added `codegen_units = 1` to the wrong build profile. Fixed now.

The build also works with the latest nightly, so we could possibly change the toolchain in `rust-toolchain.toml` to just be `nightly`.

I also needed to change `usbd-hid` dev dependency to a git dependency. The latest nightlies break on a previous version of `proc-macro2` that the latest published release of `usbd-hid` depends on.

stappersg commented:
On Sat, Apr 18, 2026 at 06:59:30PM -0700, rmsyn wrote:
> Closed #14.

For which reason?

stappersg commented:
On Sat, Apr 18, 2026 at 10:12:57PM -0700, Geert Stappers wrote:
> On Sat, Apr 18, 2026 at 06:59:30PM -0700, rmsyn wrote:
> > Closed #14.
> 
> For which reason?

Oops, I missed that this merge request was from 2023.


Thanks for closing.
